flink-connectors/flink-connector-files/src/main/java/org/apache/flink/connector/file/src/impl/FileRecordFormatAdapter.java

```diff
@@ -37,7 +37,12 @@
 import static org.apache.flink.util.Preconditions.checkArgument;
 import static org.apache.flink.util.Preconditions.checkNotNull;
 
-/** The FormatReaderAdapter turns a {@link FileRecordFormat} into a {@link BulkFormat}. */
+/**
+ * The FormatReaderAdapter turns a {@link FileRecordFormat} into a {@link BulkFormat}.
+ *
+ * @deprecated Please use {@link StreamFormatAdapter} instead.
```

Comment:
Nit: I guess this hint is not entirely correct because users cannot use the `StreamFormatAdapter` without migrating to `StreamFormat`. In general, I am not sure whether it is strictly necessary to deprecate internal classes.

Reply:
I'd go for internal classes as well so that we will not forget to remove them when we remove the public class. Ofc, the link to `StreamFormatAdapter` is debatable but I also don't see that it's hurting anyone.